Notify/handler.ts

```diff
@@ -76,14 +84,17 @@ const canSendReminderNotification = (
  * @param fiscalCode
  * @returns a TaskEither of Error or boolean
  */
-const checkSendNotificationPermission = (isBetaTester: IsBetaTester) => (
+const checkSendNotificationPermission = (
+  isBetaTester: IsBetaTester,
+  retrieveUserProfile: UserProfileReader
+) => (
   notificationType: NotificationType,
   fiscalCode: FiscalCode
 ): TE.TaskEither<Error, boolean> =>
   match(notificationType)
     .when(isReminderNotification, _ =>
       isBetaTester(fiscalCode)
-        ? canSendReminderNotification(fiscalCode)
+        ? canSendReminderNotification(retrieveUserProfile)(fiscalCode)
```

Comment:
What about use `Boolean.fold` here ?

Reply:
We're going to remove it, so we can avoid refactor right now